source/extensions/filters/udp/udp_proxy/udp_proxy_filter.cc

```diff
@@ -311,6 +315,14 @@ UdpProxyFilter::ActiveSession::~ActiveSession() {
 
   disableAccessLogFlushTimer();
 
+  for (auto& active_read_filter : read_filters_) {
+    active_read_filter->read_filter_->onSessionComplete();
+  }
+
+  for (auto& active_write_filter : write_filters_) {
+    active_write_filter->write_filter_->onSessionComplete();
+  }
+
```

Comment:
Is it possbile to delete read/write_filters here and add a ENVOY_BUG in session destructor to make sure onSessionComplete() has been called?

Reply:
To make sure onSessionComplete() is always called.

Reply:
I've added on_session_complete_called_ bool and ENVOY_BUG to make sure that onSessionComplete() is always called.
Can you, please, take a look and let me know if deleting filters is still needed?

Reply:
That would also work. Thanks!